src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -96,30 +162,15 @@ def query_trace_data(trace_id, params):
     ]
 
 
-class OrganizationEventsTraceEndpointBase(OrganizationEventsV2EndpointBase):
-    def has_feature(self, organization, request):
+# TODO(wmak): remove ignore once the base is typed
+class OrganizationEventsTraceEndpointBase(OrganizationEventsV2EndpointBase):  # type: ignore
+    def has_feature(self, organization: Organization, request: HttpRequest) -> Any:
         return features.has(
             "organizations:trace-view-quick", organization, actor=request.user
         ) or features.has("organizations:trace-view-summary", organization, actor=request.user)
 
-    def serialize_event(self, event, parent, generation=None):
-        return {
-            "event_id": event["id"],
-            "span_id": event["trace.span"],
-            "transaction": event["transaction"],
-            "transaction.duration": event["transaction.duration"],
-            "transaction.op": event["transaction.op"],
-            "project_id": event["project.id"],
-            "project_slug": event["project"],
-            "parent_event_id": parent,
-            # Avoid empty string for root events
-            "parent_span_id": event["trace.parent_span"] or None,
-            # Can be None on the light trace when we don't know the parent
-            "generation": generation,
-            "errors": [],
-        }
-
-    def serialize_error(self, event):
+    @staticmethod
+    def serialize_error(event: SnubaEvent) -> TraceError:
         return {
```

Comment:
TraceError says all of its values are strings. is project_id really a string? may consider TypedDict

Reply:
Similarly "parent_span_id" is actually an `Optional[str]`. and "errors" is a list. So the typing doesn't hold up here.